Jetpack Social: Add Jetpack Social no connection view to post settings

wargcm opened the pull request:
See #20784

## Description

- Adds a reusable no social connections view
- Uses the new view on the post settings screen when there are no social connections

## Screenshots

| Light | Dark |
|------|------|
| ![Simulator Screen Shot - Light](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/83c170cd-03d2-46cd-b617-1b3fc0800255) | ![Simulator Screen Shot - Dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/8239ef60-a8bd-4e91-a305-22a67dc554f3) |

## Testing

> **Note**
> At this time, I'm unsure of whether the white border around the icons is intended when in dark mode.

To test:
- Use an account which has no social connections
- Launch Jetpack and login
- Create a blog post or edit one
- On the editor, open the menu (`...`) in the top right
- Tap on `Post Settings`
- **Verify** the no connections view matches (minus the Twitter icon): Th1ahHKq53k5JT1PNDMavY-fi-849_12349
- Switch to a self-hosted site that does not have Jetpack Social
- Repeat the above steps to get to the `Post Settings` screen
- **Verify** the no connections UI does **NOT** appear

## Regression Notes
1. Potential unintended areas of impact
Post settings screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [X] Portrait and landscape orientations.
- [X] Light and dark modes.
- [X] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [X] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

wargcm committed:
Add Jetpack Social no connection view to post settings

dvdchr commented on WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:
```diff
@@ -0,0 +1,87 @@
+    var body: some View {
+        VStack(alignment: .leading, spacing: 12.0) {
+            HStack(spacing: -6.0) {
+                iconImage("icon-tumblr")
+                iconImage("icon-facebook")
+                iconImage("icon-linkedin")
```
I think we should update this to be dynamic based on stored `PublicizeService` objects. Perhaps you've already planned to address this separately? Either way is fine by me.

wargcm replied:
Yeah, we will have to make a couple changes to support that. Off the top of my head we'll need:

- Add the service call somewhere so the `PublicizeService` objects are always there. Probably the dashboard? We'll have the no connections view there as well.
- Someway to download the icons. iOS 15 has `AsyncImage` for SwiftUI but we still support iOS 14, so we'll need to create some sort of image view that downloads the icons.

Since these are larger changes, I'll make them in a separate PR.

dvdchr commented on WordPress/Classes/ViewRelated/Jetpack/Social/JetpackSocialNoConnectionView.swift:
```diff
@@ -0,0 +1,87 @@
+    func iconImage(_ image: String) -> some View {
+       Image(image)
+            .resizable()
+            .frame(width: 32.0, height: 32.0)
```
Should we add `.clipShape(Circle())` to force any image here to be displayed in a circular container? I'm not sure if we're going to fall back to the icon URL returned from the `external-services` API, but the images returned from there are square-shaped. More context: p1686079466982639-slack-C05362VSXFU

wargcm replied:
Sure, I can add that 👍. It's in: https://github.com/wordpress-mobile/WordPress-iOS/pull/20808/commits/af1c70a42384a06c897974699259a2a97021b2d5

dvdchr reviewed (approved):
Tested the UI in various configurations: accessibility sizes, iPad, landscape, and also tested VoiceOver. I'm wondering if it's better to group the image icons and apply a single accessibility label for them, though 🤔.

As for the icons, I know that you've exported them from Figma, but I don't think there should be a white border surrounding the icons. However, when they no longer have a border, you might need to manually add a 1-2px border with a color that matches the cell's background. In any case, I'm shamelessly pinging @osullivanchris for your help on removing the extra borders around the icon. 😄

Anyways, overall this is LGTM. Feel free to address my comment in this PR or the next.

wargcm committed:
Clip the Jetpack Social no connection images to be circles

wargcm committed:
Update social icon images to remove the white borders

wargcm committed:
Add an accessibility label to the social media icons

wargcm commented:
@dvdchr I've updated the icons with the new ones from Figma and added a border in https://github.com/wordpress-mobile/WordPress-iOS/pull/20808/commits/d0d1bc7f05fac8fd28570e3a1ff0c2cd74d4f19c. Here are a couple screenshots:

| Light | Dark |
|------|------|
| ![Screenshot Light](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/a2931589-74f2-406c-89d4-fe685c32c6d1) | ![Screenshot Dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/89277395-7f27-4847-a485-feaa41890af5) |

Additionally, I added an accessibility label to the icons in https://github.com/wordpress-mobile/WordPress-iOS/pull/20808/commits/dcf38fa44bc0e71e4f7d0adcec08979260d55927. I made it pretty generic so I was curious what your thoughts are on it. We could potentially make it a dynamic label too listing out each of the services.

dvdchr commented:
> I made it pretty generic so I was curious what your thoughts are on it. We could potentially make it a dynamic label too listing out each of the services.

@wargcm, the generic accessibility label looks good to me. I could be wrong, but my thinking is that the icons are more of a sugar to make the "no connection" section more attractive, rather than being informative of what services we're supporting. Plus, when they navigate to the Social screen via "Connect your profiles", we should have a proper VoiceOver for each of the services.

dvdchr reviewed (approved):
LGTM!
